core/src/tvu.rs

```diff
@@ -292,16 +292,18 @@ impl Tvu {
             tower_storage,
         );
 
-        let warm_quic_cache_service = if connection_cache.use_quic() {
-            Some(WarmQuicCacheService::new(
-                connection_cache.clone(),
-                cluster_info.clone(),
-                poh_recorder.clone(),
-                exit.clone(),
-            ))
-        } else {
-            None
-        };
+        let warm_quic_cache_service =
+            connection_cache
+                .filter(|cache| cache.use_quic())
+                .map(|cache| {
+                    WarmQuicCacheService::new(
+                        cache.clone(),
+                        cluster_info.clone(),
+                        poh_recorder.clone(),
+                        exit.clone(),
+                    )
+                });
+
```

Comment:
```suggestion
        let connection_cache_use_quic = connection_cache
            .map(|cc| cc.use_quic())
            .unwrap_or(false);
        let warm_quic_cache_service = if connection_cache_use_quic {
            Some(WarmQuicCacheService::new(
                connection_cache.clone(),
                cluster_info.clone(),
                poh_recorder.clone(),
                exit.clone(),
            ))
        } else {
            None
        };
```

to keep the bp diff small. we can refactor in a master-only pr

Reply:
No -- we have never done cache warming for tpu forward.